stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/OfflineSyncFirebaseMessagingService.kt

```diff
@@ -58,41 +51,37 @@ internal class OfflineSyncFirebaseMessagingService : FirebaseMessagingService()
 
         GlobalScope.launch(DispatcherProvider.IO) {
             val cid: String = firebaseMessageParser.parse(message).let { "${it.channelType}:${it.channelId}" }
-            if (ChatDomain.isInitialized) {
-                performSync(ChatDomain.instance(), cid)
-
-                if (ChatClient.isInitialized) {
-                    ChatClient.instance()
-                        .onMessageReceived(message)
-                }
-            }
-
-            if (ChatDomain.isInitialized.not() || ChatClient.isInitialized.not()) {
+            if (ChatDomain.isInitialized && ChatClient.isInitialized) {
+                performSync(ChatDomain.instance(), cid, ChatClient.instance(), message)
+            } else {
                 val syncConfig = syncModule.encryptedBackgroundSyncConfigStore.get()
-
                 syncConfig?.let {
                     val config = it
                     val user = User(id = config.userId)
-                    val client = initClient(
-                        this@OfflineSyncFirebaseMessagingService,
-                        user,
-                        config.userToken,
-                        config.apiKey
-                    )
+                    val client = initClient(this@OfflineSyncFirebaseMessagingService, config.apiKey)
                     val domain = initDomain(user, client)
-                    performSync(domain, cid)
-                    client.onMessageReceived(message)
+                    logger.logD("Starting the sync, config: $syncConfig")
+                    performSync(domain, cid, client, message)
                 }
             }
-
             stopForeground(true)
             stopSelf()
         }
     }
 
-    private fun performSync(domain: ChatDomain, cid: String) {
-        domain.apply {
-            useCases.replayEventsForActiveChannels(cid).execute()
+    private fun performSync(
+        domain: ChatDomain,
+        cid: String,
+        client: ChatClient,
+        message: RemoteMessage,
+    ) {
+        domain.useCases.replayEventsForActiveChannels(cid).enqueue {
+            if (it.isSuccess) {
+                logger.logD("sync success.")
+            } else {
+                logger.logD("sync failed")
+            }
+            client.onMessageReceived(message)
```

Comment:
When app process was killed fails loading channel here
```
    private fun loadRequiredData(channelType: String, channelId: String, messageId: String) {
        val getMessage = client.getMessage(messageId)
        val getChannel = client.queryChannel(channelType, channelId, QueryChannelRequest())

        getChannel.zipWith(getMessage).enqueue { result ->
            if (result.isSuccess) {
                val (channel, message) = result.data()

                handler.getDataLoadListener()?.onLoadSuccess(channel, message)
                onRequiredDataLoaded(channel, message)
            } else {
                logger.logE("Error loading required data: ${result.error().message}", result.error())
                handler.getDataLoadListener()?.onLoadFail(messageId, result.error())
                showErrorCaseNotification()
            }
        }
    }
```
with `ChatNetworkError http status -1, stream error code 1007: No defined token. Check if client.setUser was called and finished`